flang/test/Lower/intrinsic-procedures/lbound.f90

```diff
@@ -2,69 +2,51 @@
 
 
 ! CHECK-LABEL: func @_QPlbound_test(
-! CHECK-SAME:  %[[VAL_0:.*]]: !fir.box<!fir.array<?x?xf32>>,
-! CHECK-SAME:  %[[VAL_1:.*]]: !fir.ref<i64>,
-! CHECK-SAME:  %[[VAL_2:.*]]: !fir.ref<i64>) {
+! CHECK:  %[[VAL_0:.*]] = fir.load %arg1 : !fir.ref<i64>
+! CHECK:  %[[VAL_1:.*]] = fir.rebox %arg0 : (!fir.box<!fir.array<?x?xf32>>) -> !fir.box<!fir.array<?x?xf32>>
 subroutine lbound_test(a, dim, res)
   real, dimension(:, :) :: a
   integer(8):: dim, res
-! CHECK:         %[[VAL_3:.*]] = arith.constant 1 : i64
-! CHECK:         fir.store %[[VAL_3]] to %[[VAL_2]] : !fir.ref<i64>
+! CHECK:         %[[VAL_2:.*]] = fir.address_of(
+! CHECK:         %[[VAL_3:.*]] = fir.convert %[[VAL_1]] : (!fir.box<!fir.array<?x?xf32>>) -> !fir.box<none>
+! CHECK:         %[[VAL_4:.*]] = fir.convert %[[VAL_0]] : (i64) -> i32
+! CHECK:         %[[VAL_5:.*]] = fir.convert %[[VAL_2]]
+! CHECK:         %[[VAL_6:.*]] = fir.call @_FortranALboundDim(%[[VAL_3]], %[[VAL_4]], %[[VAL_5]], %{{.*}}) : (!fir.box<none>, i32, !fir.ref<i8>, i32) -> i64
+! CHECK:         fir.store %[[VAL_6]] to %arg2 : !fir.ref<i64>
   res = lbound(a, dim, 8)
 end subroutine
 
 ! CHECK-LABEL: func @_QPlbound_test_2(
-! CHECK-SAME:  %[[VAL_0:.*]]: !fir.box<!fir.array<?x?xf32>>,
-! CHECK-SAME:  %[[VAL_1:.*]]: !fir.ref<i64>,
-! CHECK-SAME:  %[[VAL_2:.*]]: !fir.ref<i64>) {
+! CHECK:  %[[VAL_0:.*]] = fir.convert %{{.*}} : (i64) -> index
+! CHECK:  %[[VAL_1:.*]] = fir.convert %{{.*}} : (i64) -> index
```

Comment:
Can you just add a CHECK that this value comes from `arith.constant 2` ? I think it is a crucial point of this test.

Reply:
Will do.   Thanks for pointing this out.